src/Init/Data/Int/LemmasAux.lean

```diff
@@ -151,4 +151,46 @@ theorem bmod_bmod_of_dvd {a : Int} {n m : Nat} (hnm : n ∣ m) :
   obtain ⟨k, rfl⟩ := hnm
   simp [Int.mul_assoc]
 
+theorem bmod_eq_of_le_of_lt {x : Int} {y : Nat} (hle : -y ≤ x * 2) (hlt : x * 2 < y) :
+    x.bmod y = x := by
+  simp only [Int.bmod_def]
+  rcases x
+  · rw [Int.emod_eq_of_lt (by simp only [ofNat_eq_coe]; omega) (by omega)]; omega
+  · rw [Int.emod_eq_add_self_emod, Int.emod_eq_of_lt (by omega) (by omega)]; omega
+
+theorem mul_le_mul_self_of_natAbs_le {x y : Int} {s : Nat} (hx : x.natAbs ≤ s) (hy : y.natAbs ≤ s) :
+    x * y ≤ s * s := by
```

Comment:
This generalizes to
```suggestion
theorem mul_le_mul_of_natAbs_le {x y : Int} {s t : Nat} (hx : x.natAbs ≤ s) (hy : y.natAbs ≤ t) :
    x * y ≤ s * t := by
```
and I suspect so does `neg_mul_self_le_mul` below.

Reply:
@TwoFX suggested the more specific lemma as a *corollary* of the more general fact, since it was what we needed to control the particular situation at hand :) I am still mildly in favor of keeping it specific, especially since the name `mul_le_mul_self_of_natAbs_le` communicates what bound is being established at the use site.

As a compromise, I'm happy to have the general version (`mul_le_mul_of_natAbs_le`), from which we immediately derive the more specific corollary (`mul_le_mul_self_of_natAbs_le`), which we then use in the proof?

Reply:
I don't follow. `mul_le_mul_of_natAbs_le` is strictly more general, has a very natural statement (being one of several good answer to the question "what is the correct statement of `Nat.mul_le_mul` for integers?"), your use site required no modification to switch to the more general form, and there is no corresponding `Nat.mul_le_mul_self`, so I don't see a reason to introduce `mul_le_mul_self_of_natAbs_le`.

Reply:
thank you for the comments, I left the general version.